analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.cs

```diff
@@ -29,5 +29,101 @@ public class PublicMethodArgumentsShouldBeCheckedForNull : SymbolicRuleCheck
 
     protected override DiagnosticDescriptor Rule => S3900;
 
-    public override bool ShouldExecute() => false;
+    public override bool ShouldExecute()
+    {
+        return IsAccessibleFromOtherAssemblies(Node)
+            && (IsRelevantMethod(Node) || IsRelevantPropertyAccessor(Node));
+
+        static bool IsRelevantMethod(SyntaxNode node) =>
+            node is BaseMethodDeclarationSyntax { } method
+            && MethodDereferencesArguments(method);
+
+        static bool IsRelevantPropertyAccessor(SyntaxNode node) =>
+            node is AccessorDeclarationSyntax accessor
+            && IsPropertyAccessorAccessibleFromOtherAssemblies(accessor.Modifiers);
+
+        static bool IsAccessibleFromOtherAssemblies(SyntaxNode node) =>
+            node.AncestorsAndSelf().OfType<MemberDeclarationSyntax>().FirstOrDefault() is { } containingMember
+            && node.Ancestors().OfType<BaseTypeDeclarationSyntax>().FirstOrDefault() is { } containingType
+            && IsMemberAccessibleFromOtherAssemblies(containingMember.Modifiers(), containingType)
+            && IsTypeAccessibleFromOtherAssemblies(containingType.Modifiers);
+
+        static bool IsMemberAccessibleFromOtherAssemblies(SyntaxTokenList modifiers, BaseTypeDeclarationSyntax containingType) =>
+            IsPublicOrProtectedOrProtectedInternal(modifiers)
+            || (containingType is InterfaceDeclarationSyntax && HasNoDeclaredAccessibilityModifier(modifiers));
+
+        static bool IsPropertyAccessorAccessibleFromOtherAssemblies(SyntaxTokenList modifiers) =>
+            IsPublicOrProtectedOrProtectedInternal(modifiers)
+            || HasNoDeclaredAccessibilityModifier(modifiers);
+
```

Comment:
This doesn't need to be extracted. Embed it into the call site (this will invalidate the comment about the callsite fitting into a single line)

Reply:
Removed it. Now using the semantic model for checking accessibility.